nextest-runner/src/reporter/displayer.rs

```diff
@@ -29,12 +29,172 @@ use serde::Deserialize;
+        }
+    }
+
+    /// Which output streams should be output at the end
+    ///
+    /// # Returns
+    /// Returns `None` when no output streams should be output
+    pub fn display_output_final(&self) -> Option<DisplayOutput> {
+        let stdout = self.stdout.map_or(false, |t| t.is_final());
+        let stderr = self.stderr.map_or(false, |t| t.is_final());
+        if stdout || stderr {
+            Some(DisplayOutput { stdout, stderr })
+        } else {
+            None
+        }
+    }
+}
+
+impl<'de> Deserialize<'de> for TestOutputDisplayStreams {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: serde::de::Deserializer<'de>,
+    {
+        deserializer.deserialize_any(TestOutputDisplayStreamsVisitor)
+    }
+}
+
+struct TestOutputDisplayStreamsVisitor;
+impl<'de> serde::de::Visitor<'de> for TestOutputDisplayStreamsVisitor {
+    type Value = TestOutputDisplayStreams;
+
+    fn expecting(&self, formatter: &mut Formatter) -> fmt::Result {
+        formatter.write_str(
+            "a string with 'never', 'immediate', 'immediate-final' or 'final',\
+or a map with 'stdout' and/or 'stderr' as keys and the preceding values",
+        )
+    }
+
+    fn visit_str<E>(self, v: &str) -> Result<Self::Value, E>
+    where
+        E: serde::de::Error,
+    {
+        // the input is a string so we expect a TestOutputDisplay value
+        match v {
+            "never" => Ok(TestOutputDisplayStreams::create_never()),
+            "immediate" => Ok(TestOutputDisplayStreams::create_immediate()),
+            "immediate-final" => Ok(TestOutputDisplayStreams::create_immediate_final()),
+            "final" => Ok(TestOutputDisplayStreams::create_final()),
+            _ => Err(E::invalid_value(
+                serde::de::Unexpected::Str(v),
+                &"unrecognized value, expected 'never', 'immediate', 'immediate-final' or 'final'\
+or a map with 'stdout' and/or 'stderr' as keys and the preceding values",
+            )),
+        }
+    }
+
+    fn visit_map<A>(self, mut map: A) -> Result<Self::Value, A::Error>
+    where
+        A: serde::de::MapAccess<'de>,
+    {
+        use serde::de::Error;
+
+        // the input is a map, so we expect stdout and/or stderr as keys and TestOutputDisplay values
+        // as the value
+        let mut stdout = None;
+        let mut stderr = None;
+        while let Some((key, value)) = map.next_entry::<&str, &str>()? {
+            match key {
+                "stdout" => {
+                    if stdout.is_some() {
+                        return Err(A::Error::duplicate_field("stdout"));
+                    }
+                    stdout = Some(TestOutputDisplay::from_str(value).map_err(|_| {
+                        A::Error::invalid_value(
+                            serde::de::Unexpected::Str(value),
+                            &"'never', 'immediate', 'immediate-final', or 'final'",
+                        )
+                    })?);
```

Comment:
As the warnings suggest, for a manual deserialize impl it would be good to have tests for all the error cases. (I know you haven't written tests yet, just wanted to mention this.)